src/iuse.cpp

```diff
@@ -4167,21 +4168,13 @@ int iuse::gasmask( player *p, item *it, bool t, const tripoint &pos )
             const field &gasfield = g->m.field_at( pos );
             for( auto &dfield : gasfield ) {
                 const field_entry &entry = dfield.second;
-                const field_id fid = entry.get_field_type();
-                switch( fid ) {
-                    case fd_smoke:
-                        it->set_var( "gas_absorbed", it->get_var( "gas_absorbed", 0 ) + 12 );
-                        break;
-                    case fd_tear_gas:
-                    case fd_toxic_gas:
-                    case fd_gas_vent:
-                    case fd_smoke_vent:
-                    case fd_relax_gas:
-                    case fd_fungal_haze:
-                        it->set_var( "gas_absorbed", it->get_var( "gas_absorbed", 0 ) + 15 );
-                        break;
-                    default:
-                        break;
+                const field_type_id fid = entry.get_type();
+                if( fid == fd_smoke ) {
+                    it->set_var( "gas_absorbed", it->get_var( "gas_absorbed", 0 ) + 12 );
+                }
+                if( fid == fd_tear_gas || fid == fd_toxic_gas || fid == fd_gas_vent ||
+                    fid == fd_smoke_vent || fid == fd_relax_gas || fid == fd_fungal_haze ) {
+                    it->set_var( "gas_absorbed", it->get_var( "gas_absorbed", 0 ) + 15 );
```

Comment:
Ditto here, why scramble the `switch`?

Reply:
I will refactor this later. The approach would be to have some nodes in field type definitions which determine processing of fields, i.e.:

Field types `fd_tear_gas`, `fd_toxic_gas`, `fd_gas_vent`, `fd_smoke_vent`, `fd_relax_gas` and `fd_fungal_haze` (and any new similar fields) will have some node in json (e.g. `"field_processing_rule_values": { { "item_gas_absorbed_delta": 15 } }`) which will indicate they would need to go with ```it->set_var( "gas_absorbed", it->get_var( "gas_absorbed", 0 ) + ft.field_processing_rule_values.item_gas_absorbed_delta );```.